BridgeAppSDK/SBAProfileItem.swift

```diff
@@ -523,7 +557,170 @@ open class SBAUserDefaultsProfileItem: SBAProfileItemBase {
 
 }
 
-open class SBAFullNameProfileItem: SBAKeychainProfileItem, SBANameDataSource {
+enum SBAProfileParticipantSourceKey: String {
+    case firstName
+    case lastName
+    case email
+    case externalId
+    case notifyByEmail
+    case sharingScope
+    case dataGroups
+}
+
+open class SBAStudyParticipantProfileItem: SBAStudyParticipantCustomAttributesProfileItem {
+    public static var studyParticipant: SBBStudyParticipant?
+    
+    override open func storedValue(forKey key: String) -> Any? {
+        guard let studyParticipant = SBAStudyParticipantProfileItem.studyParticipant
+            else {
+                assertionFailure("Attempting to read \(key) (\(profileKey)) on nil SBBStudyParticipant")
+                return nil
+        }
+        guard !(key.hasPrefix("attributes.")) else {
+            let pathComponents = key.components(separatedBy: ".")
+            let attributeComponents = pathComponents[1...pathComponents.count - 1]
+            guard attributeComponents.count > 0 else {
+                assertionFailure("Error reading \(key) (\(profileKey)): \(key) is not a valid SBBStudyParticipant keypath")
+                return nil
+            }
+            let attributeKey = attributeComponents.joined(separator:".")
```

Comment:
I put this in (with this inheritance) so that a custom attribute key (preferredName) could fall back to a standard StudyParticipant key (firstName). I'll apply the simplification.